homeassistant/components/intellifire/sensor.py

```diff
@@ -0,0 +1,146 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Any
+
+from intellifire4py import IntellifirePollData
+import pytz
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IntellifireDataUpdateCoordinator
+from .const import DOMAIN
+
+
+class IntellifireSensor(CoordinatorEntity, SensorEntity):
+    """Define a generic class for Sensors."""
+
+    # Define types
+    coordinator: IntellifireDataUpdateCoordinator
+    entity_description: IntellifireSensorEntityDescription
+    _attr_attribution = "Data provided by unpublished Intellifire API"
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        description: IntellifireSensorEntityDescription,
+    ) -> None:
+        """Init the sensor."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+
+        # Set the Display name the User will see
+        self._attr_name = f"Fireplace {description.name}"
+        self._attr_unique_id = f"{description.key}_{coordinator.api.data.serial}"
+        # Configure the Device Info
+        self._attr_device_info = self.coordinator.device_info
+
+    @property
+    def native_value(self) -> int | str:
+        """Return the state."""
+        return self.entity_description.value_fn(self.coordinator.api.data)
+
+
+class IntellifireTimeSensor(IntellifireSensor):
+    """Define a sensor that takes into account timezone."""
+
+    @property
+    def native_value(self) -> Any:
+        """Return a datetime value."""
+        seconds_offset = int(
+            self.entity_description.value_fn(self.coordinator.api.data)
+        )
+        # If disabled return None - else return a timestamp with correct TZ info
+        if seconds_offset == 0:
+            return None
+
+        return datetime.now().replace(
+            tzinfo=pytz.timezone(self.hass.config.time_zone)
+        ) + timedelta(seconds=seconds_offset)
+
+
+@dataclass
+class IntellifireSensorRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    # Although sensors could have a variety of different return values,
+    # all the ones below are only returning ints
+    value_fn: Callable[[IntellifirePollData], int | str]
+
+
+@dataclass
+class IntellifireSensorEntityDescription(
+    SensorEntityDescription, IntellifireSensorRequiredKeysMixin
+):
+    """Describes a sensor sensor entity."""
+
+    entity_class: type[IntellifireSensor] = IntellifireSensor
+
```

Comment:
```suggestion
    entity_class: type[IntellifireSensor] = IntellifireSensor

```

Reply:
Did you mean something else here? This is basically a zero change change? :)

Reply:
It's a removal => drop the line.